### ISSUE TITLE
[INSTALL]: Need spack-stack/1.6.0 on S4 for Rocky 8

### ISSUE THREAD

**Issue by InnocentSouopgui-NOAA** [JCSDA/spack-stack#1324]
### Package name

spack-stack

### Package version/tag

1.6.0

### Build options

current

### Installation timeframe

ASAP.
S4 was just upgraded to Rocky 8 and Spack-stack is needed for users to keep their experiments going.

### Other information

Essentially need spack-stack-1.6.0 installed for the new OS, Rocky 8

**Comment by climbfuji** [comment]
JCSDA is in charge of S4 (@srherbener, @ashley314). In any case, this has to wait until the maintenance/upgrade is complete. As per Jesse's email from yesterday, users can start testing the updated system, but it's not yet ready for a roll out of all spack-stack versions.

My suggestion would be for JCSDA to install spack-stack-1.8.0 with Intel and GNU for testing first, and once the configuration passed acceptance testing move on to reinstall the legacy version of spack-stack-1.6.0 with Intel.

**Comment by InnocentSouopgui-NOAA** [comment]
@climbfuji, thank you.
It will be great great to have a testing install of spack-stack.
I am in charge of global workflow on S4, and I will greatly appreciate having that testing version, to start preparing global workflow and its components.

**Comment by climbfuji** [comment]
@InnocentSouopgui-NOAA I talked with JCSDA, I will go ahead and update the site config/install spack-stack-1.8.0 with Intel first. Then I'll look into the GNU stack.

**Comment by InnocentSouopgui-NOAA** [comment]
Any update on the install of spack-stack 1.6.0 on S4?
It will be helpful so that users of that system can have their work going on after the OS upgrade.

Thank you.

**Comment by climbfuji** [comment]
Tagging @srherbener

**Comment by wx20jjung** [comment]
As an S4 user, it would be helpful if we could get a timeline as to when  this issue will be resolved so we can begin our global-workflow projects again.
Jim

**Comment by InnocentSouopgui-NOAA** [comment]
Any update?
S4 user are waiting to resume their research.

If there are updated instructions to install Spack Stack on S4, I can help. The instructions in the documentation are not working out-of-the-box.

Thanks.

**Comment by srherbener** [comment]
Sorry about the delay with this. I see a spack-stack-1.6.0 installation on S4 here: `/data/prod/jedi/spack-stack/spack-stack-1.6.0`. Before starting, I just want to confirm which intel compiler version is required. The current builds of spack-stack-1.6.0 used 2021.5.0, but 2021.10.0 is now available. Thanks!

**Comment by wx20jjung** [comment]
Jesse,

Would you identify the Intel compiler version we (the users) should (will)
be using.
Thanks,
Jim

**Comment by srherbener** [comment]
It looks like to me that 2021.5.0 (/opt/intel/oneapi/2022.1) is no longer available, and the only choice is 2021.10.0 (/opt/intel/oneapi/2023.2). I'll get started, but it would be good still to get the clarification.

**Comment by InnocentSouopgui-NOAA** [comment]
/opt/intel/oneapi/2023.2 is the compiler that will be used.
Thank you for getting to this.

**Comment by InnocentSouopgui-NOAA** [comment]
All the version from spack-stack-1.4.0 to 1.7.0, already on system are from the previous OS.
They can be archived as they won't be used in the future.

**Comment by eap** [comment]
@InnocentSouopgui-NOAA - is there a reason that it is not feasible to use spack-stack 1.8?

Generally we try to support spack-stack releases for up to a 1 year lifetime, as 1.6 is now 10 months old any work done now would simply preserve it for an additional 2 months. Without good justification (and in light of substantial resource constraints) I'd prefer we use this as an opportunty to upgrade unless we have a really strong justification for updating the 1.6 release. Side note: if 1.6 works in a single shot, I'd be supportive but I strongly suspect it will take additional site customization due to the large number of upgrades. I don't suspect this will be as simple as waiting on a build.

**Comment by climbfuji** [comment]
spack-stack-1.6.0 is used by the unified forecast system (UFS) for many components. ufs-weather-model, global-workflow, ... - we have to support this release until they move up to 1.8.0 (they will skip 1.7.0).

**Comment by eap** [comment]
@climbfuji - In that case, at least we can skip re-releasing 1.7.

**Comment by climbfuji** [comment]
Yes, correct. I'd rebuild only 1.6.0 with Intel (and a simple diff of the site configs might do the job) and only reinstall others if someone complains.

**Comment by InnocentSouopgui-NOAA** [comment]
Yes we can safely skip version 1.7. But version 1.6.0 is needed for S4 users to continue running Global workflow and its components while waiting for Global workflow to be migrated to stack spack 1.8.0

**Comment by srherbener** [comment]
I have a 1.6.0 build running on S4 (with Intel 2021.10.0) now. Here's to smooth sailing!

**Comment by InnocentSouopgui-NOAA** [comment]
> I have a 1.6.0 build running on S4 (with Intel 2021.10.0) now. Here's to smooth sailing!

Please let me know when you are done, so that I can start building other software that depend on Spack Stack.

**Comment by srherbener** [comment]
The spack-stack-1.6.0 build is completed. It is located in:

/data/prod/jedi/spack-stack/spack-stack-1.6.0/envs/ue-intel-2021.10.0

The spack-stack 1.6.0 readthedocs instructcions say to do the following to load the 1.6.0 intel 
 unified environment:
```
module purge
module use /data/prod/jedi/spack-stack/modulefiles
module load miniconda/3.9.12
module load ecflow/5.8.4
module load mysql/8.0.31

module use /data/prod/jedi/spack-stack/spack-stack-1.6.0/envs/unified-env/install/modulefiles/Core
module load stack-intel/2021.5.0
module load stack-intel-oneapi-mpi/2021.5.0
module load stack-python/3.10.13
module unuse /opt/apps/modulefiles/Compiler/intel/non-default/22
module unuse /opt/apps/modulefiles/Compiler/intel/22
module available
```

I would follow these instructions except to replace the intel related lines to use the new compiler. This should work:
```
module purge
module use /data/prod/jedi/spack-stack/modulefiles
module load miniconda/3.9.12
module load ecflow/5.8.4
module load mysql/8.0.31

module use /data/prod/jedi/spack-stack/spack-stack-1.6.0/envs/ue-intel-2021.10.0/install/modulefiles/Core
module load stack-intel/2021.10.0
module load stack-intel-oneapi-mpi/2021.10.0
module load stack-python/3.10.13
module unuse /opt/apps/modulefiles/Compiler/intel/23
module available
```

Where should this adjustment be documented? On the spack-stack wiki?

**Comment by srherbener** [comment]
Yesterday it was noted that we also need to rebuild the `gsi-addon` environment. @DavidHuber-NOAA supplied me with instructions, and I'll work on that now.

**Comment by climbfuji** [comment]
Thanks very much, Steve. I think the Wiki is a great place to document these updates. I created https://github.com/JCSDA/spack-stack/wiki/Post%E2%80%90release-updates-for-spack%E2%80%90stack%E2%80%901.6.0 and linked it to the top page: https://github.com/JCSDA/spack-stack/wiki

**Comment by InnocentSouopgui-NOAA** [comment]
Thank you so much for the updates.

**Comment by InnocentSouopgui-NOAA** [comment]
@srherbener ,

Can you also rebuild the `upp-addon` environment, it has the correct version of libraries needed for the UFS model. 

Thanks.

**Comment by srherbener** [comment]
> @srherbener ,
> 
> Can you also rebuild the `upp-addon` environment, it has the correct version of libraries needed for the UFS model.
> 
> Thanks.

Okay. I'll need build instructions for upp-addon.

@DavidHuber-NOAA, it looks like the instructions you gave me for gsi-addon are working. Thanks! Do you also have instructions for upp-addon? Do I simply substitute "gsi-addon" with "upp-addon" in the gsi-addon instructions? Thanks!

**Comment by DavidHuber-NOAA** [comment]
@AlexanderRichert-NOAA @srherbener I haven't ever built the upp-addon environment, and looking in configs/templates, I do not see an obvious environment to build off of.  Alex, do you know which template was used to build the upp-addon environment?

**Comment by InnocentSouopgui-NOAA** [comment]
Will it make sense to copy `envrepo` (`/data/prod/jedi/spack-stack/spack-stack-1.6.0/envs/upp-addon-env/envrepo`) from the previous build?

**Comment by srherbener** [comment]
The gsi-addon environment is finished and ready for trying: /data/prod/jedi/spack-stack/spack-stack-1.6.0/envs/gsi-addon-ue-intel-2021.10.0

Not sure how to use it, so someone needs to come up with the recipe for loading the module environment. I suspect it's the same as the recipe for the main intel env, with a couple more module loads added in.

**Comment by srherbener** [comment]
> Thanks very much, Steve. I think the Wiki is a great place to document these updates. I created https://github.com/JCSDA/spack-stack/wiki/Post%E2%80%90release-updates-for-spack%E2%80%90stack%E2%80%901.6.0 and linked it to the top page: https://github.com/JCSDA/spack-stack/wiki

I've updated the post release notes for 1.6.0, S4 with the instructions on the new way to load the intel unified environment.

**Comment by eap** [comment]
@InnocentSouopgui-NOAA  - We are reviewing this issue now, looks like the `upp-addon` was not part of the original 1.6 release process for the S4 platform which lead to this getting left out of the re-release. This issue has been consuming a great deal of our support resources for an older release that most jedi devs are not using.

I think the right move here is for you to get jedipara role access so you can manage and update the 1.6 release as is necessary for your continued work. @BenjaminTJohnson might be able to help with this?

**Comment by InnocentSouopgui-NOAA** [comment]
@eap, Thank you for your assistance with getting S4 back in use.
I will request the jedipara role access.

**Comment by BenjaminTJohnson** [comment]
@InnocentSouopgui-NOAA I will request the jedipara role for you, what's your S4 username?

**Comment by InnocentSouopgui-NOAA** [comment]
@BenjaminTJohnson,
I checked and I already have the access. No action is needed.
My username on S4 is isouopgui.

**Comment by InnocentSouopgui-NOAA** [comment]
I saw the `fms-2024.01` environment on Jet that provides `fms-2024.01` for the UFS model.
How can I install that on S4?

I tried by copying envrepo from Jet but I got the error bellow at concretize step.
```
==> Error: concretization failed for the following reasons:

   1. Cannot satisfy 'fms@2024.01'
   2. Cannot satisfy 'fms@2024.01'
        required because fms@2024.01%intel requested from CLI
```

**Comment by srherbener** [comment]
@InnocentSouopgui-NOAA can this issue be closed, or is there remaining work to be done? Thanks!

**Comment by eap** [comment]
@InnocentSouopgui-NOAA is working on creating a chained environment with `fms@2024.01` but I wonder if that should be tracked in https://github.com/JCSDA/spack-stack/issues/1215 (although that issue is old and I think tracked the install prior to the rocky 8 upgrade?).

Can I propose we close this and track further work in a new issue? My understanding is that we have a working spack-stack 1.6 install on S4+rocky8.

Please re-open if I'm missing something important here.

**Comment by InnocentSouopgui-NOAA** [comment]
@eap ,
I agree, It is a very old issue.
I will open a new one.
